extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/discovery/AstroDiscoveryService.java

```diff
@@ -31,42 +36,77 @@
     /**
-     * Creates a AstroDiscoveryService with disabled autostart.
+     * Creates a AstroDiscoveryService with enabled autostart.
      */
     public AstroDiscoveryService() {
-        super(new HashSet<>(Arrays.asList(new ThingTypeUID(BINDING_ID, "-"))), DISCOVER_TIMEOUT_SECONDS, false);
+        super(new HashSet<>(Arrays.asList(new ThingTypeUID(BINDING_ID, "-"))), DISCOVER_TIMEOUT_SECONDS, true);
+
+        scheduledExecutor = ExpressionThreadPoolManager.getExpressionScheduledPool("astro");
     }
 
     @Override
     protected void startScan() {
         logger.debug("Starting Astro discovery scan");
-
         PointType location = locationProvider.getLocation();
-
         if (location == null) {
             logger.debug("LocationProvider.getLocation() is not set -> Will not provide any discovery results");
             return;
         }
+        createResults(location);
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        Expression expression = null;
+        try {
+            expression = new CronExpression(
+                    CronHelper.createCronForRepeatEverySeconds(LOCATION_CHANGED_CHECK_INTERVAL));
+        } catch (ParseException e) {
+            return;
+        }
+        if (expression != null && backgroundJob == null) {
+            AstroDiscoveryLocationChangedTask job = new AstroDiscoveryLocationChangedTask(this);
+            scheduledExecutor.schedule(job, expression);
+            logger.info("Scheduled astro location-changed job every {} seconds", LOCATION_CHANGED_CHECK_INTERVAL);
+            backgroundJob = job;
+        }
+    }
 
-        ThingUID sunThing = new ThingUID(AstroBindingConstants.THING_TYPE_SUN, LOCAL);
-        ThingUID moonThing = new ThingUID(AstroBindingConstants.THING_TYPE_MOON, LOCAL);
+    @Override
+    protected void stopBackgroundDiscovery() {
+        if (backgroundJob != null) {
+            scheduledExecutor.remove(backgroundJob);
+            backgroundJob = null;
+        }
+    }
 
+    public void createResults(PointType location) {
         String propGeolocation;
         if (location.getAltitude() != null) {
             propGeolocation = String.format("%s,%s,%s", location.getLatitude(), location.getLongitude(),
                     location.getAltitude());
         } else {
             propGeolocation = String.format("%s,%s", location.getLatitude(), location.getLongitude());
         }
-        thingDiscovered(DiscoveryResultBuilder.create(sunThing).withLabel("Local Sun")
+        thingDiscovered(DiscoveryResultBuilder.create(SUN_THING).withLabel("Local Sun")
                 .withProperty("geolocation", propGeolocation).build());
-        thingDiscovered(DiscoveryResultBuilder.create(moonThing).withLabel("Local Moon")
+        thingDiscovered(DiscoveryResultBuilder.create(MOON_THING).withLabel("Local Moon")
                 .withProperty("geolocation", propGeolocation).build());
     }
 
+    public LocationProvider getLocationProvider() {
```

Comment:
Just completely remove `AstroDiscoveryLocationChangedTask` and use a lambda here. No need for this function either in that case.